src/variant_detection/snp_indel_detection.cpp

```diff
@@ -47,48 +136,18 @@ void detect_snp_and_indel(std::filesystem::path const & reads_filename)
             continue;
         }
 
-        // Track the current position within the read.
-        unsigned read_pos = 0;
-
-        // Step through the CIGAR string.
-        for (auto && [length, operation] : record.cigar_sequence())
-        {
-            // Skip long variants.
-            if (length >= 30) // TODO (joergi-w 30.09.2021) Use the min_length parameter here.
-            {
-                read_pos += length;
-                ref_pos += length;
-                continue;
-            }
+        // Check if we have already allocated the length of the reference sequence.
+        if (activity_vec[ref_id].empty())
+            activity_vec[ref_id].resize(std::get<0>(header.ref_id_info[ref_id])); // length of reference sequence
```

Comment:
Wouldn't this perform the resize multiple times, if you begin with several reads in a row which are just 151M? Because then `update_activity_for_record` doesn't perform any changes to the vector.

Or are redundant resizes not a problem?

Reply:
No, I disagree: This resize is only executed once per reference id. For a file with 3 references, we initialize `activity_vec` as three empty vectors `[{}{}{}]` in line 120. The first time that a new ref_id appears in a record, we resize this one to the reference length (and implicitly fill zeroes). The `empty()` check returns `false` after resize!

More background why I implement it this way: I had a sam file of the human genome with 25 reference ids. However, it contained reads only of one chromosome, and we would unnecessarily allocate all the 25 vectors, although only in a single one there is relevant data.